src/main/java/org/jenkinsci/plugins/mesos/api/LaunchCommandBuilder.java

```diff
@@ -40,8 +40,15 @@ public LaunchCommandBuilder() {}
   // We allocate extra memory for the JVM
   private static final int JVM_XMX = 32;
 
-  private static final String AGENT_COMMAND_FORMAT =
+  public static enum AgentCommandStyle {
+    Linux,
+    Windows
+  }
+
+  private static final String LINUX_AGENT_COMMAND_TEMPLATE =
       "java -DHUDSON_HOME=jenkins -server -Xmx%dm %s -jar ${MESOS_SANDBOX-.}/agent.jar %s %s -jnlpUrl %s";
+  private static final String WINDOWS_AGENT_COMMAND_TEMPLATE =
+      "java -DHUDSON_HOME=jenkins -server -Xmx%dm %s -jar %MESOS_SANDBOX%/agent.jar %s %s -jnlpUrl %s";
```

Comment:
```suggestion
      "java -DHUDSON_HOME=jenkins -server -Xmx%dm %s -jar %%MESOS_SANDBOX%%/agent.jar %s %s -jnlpUrl %s";
```
The percentage sign wasn't escaped properly, this leads to failures.